fakebook/src/app/component/notifications-view/notifications-view.component.ts

```diff
@@ -25,25 +25,33 @@ export class NotificationsViewComponent implements OnInit {
       if (this.notificationsView === false) {
         this.unreadNotifications = true;
       }
-      console.log(this.notifications);
+      //console.log(this.notifications);
     });
   }
 
   toggleNotifications(): void {
     this.notificationsView = !this.notificationsView;
-    if (this.notificationsView === true) {
-      this.unreadNotifications = false;
-      this.updateNotifications();
-    }
   }
 
-  updateNotifications(): void {
+  markAllAsRead(): void {
     const ids: string[] = [];
 
     this.notifications.forEach((element) => {
       ids.push(element.id);
     });
 
     this.notifService.setRead(ids);
+    this.unreadNotifications = false;
+    this.notifications = [];
+  }
+
+  markAsRead(notif: Notification): void {
+    this.notifService.setRead([notif.id]);
+    let index = this.notifications.indexOf(notif);
+    this.notifications = this.notifications.filter((val, index, arr) => {
```

Comment:
you don't need to declare index or arr if you aren't going to use them.
and why calculate the index in the line above if you won't use it?
use `splice` instead - search for "remove item from array javascript"